### PR TITLE
Don't use color if NO_COLOR environment variable is set

### LINKED ISSUE #1073: Globally disable color?
I cannot find the way to globally disable color in the pre-commit output. Setting only the background color to green and not changing the foreground color does not work for my terminal with the following settings in the Xt resources (as set in the `${HOME}/.Xresources` file):

````properties
Rxvt.background: black
Rxvt.foreground: deepSkyBlue
````

Is there a way? It would be great to respect https://no-color.org/ environment variable. And, while we are here, maybe understand the following git config setting:

````ini
[color]
        ui = never
````

### PR DESCRIPTION
**Pull request by geieredgar**
Resolves #1073

### CODE REVIEW

File: pre_commit/color.py [hunk 1]
```diff
@@ -48,6 +48,9 @@ def use_color(setting):
     if setting not in COLOR_CHOICES:
         raise InvalidColorSetting(setting)
 
+    if 'NO_COLOR' in os.environ:
```
Comment (asottile) [inline pre_commit/color.py]: this isn't quite right -- `NO_COLOR= git commit ...` should still have color

Comment (geieredgar) [reply pre_commit/color.py]: @asottile Ok, I pushed a new commit

File: pre_commit/color.py [hunk 2]
```diff
@@ -48,7 +48,7 @@ def use_color(setting):
     if setting not in COLOR_CHOICES:
         raise InvalidColorSetting(setting)
 
-    if 'NO_COLOR' in os.environ:
+    if 'NO_COLOR' in os.environ and os.environ['NO_COLOR']:
```
Comment (asottile) [inline pre_commit/color.py]: `if os.environ.get('NO_COLOR')` would be ~slightly simpler

**Comment by geieredgar** [comment]
Ok, I will do that

File: pre_commit/color.py (hunk 2 above)
Comment (geieredgar) [reply pre_commit/color.py]: Ok, I will change that.

File: tests/color_test.py [hunk 3]
```diff
@@ -50,3 +51,20 @@ def test_use_color_tty_without_color_support():
 def test_use_color_raises_if_given_shenanigans():
     with pytest.raises(InvalidColorSetting):
         use_color('herpaderp')
+
+
+def test_no_color_env_unset():
+    with mock.patch.dict(os.environ):
```
Comment (asottile) [inline tests/color_test.py]: `clear=True` is what you want here I believe

Comment (geieredgar) [reply tests/color_test.py]: Wouldn't that remove all entries from `os.environ`? It would probably still work, though.

Comment (asottile) [reply tests/color_test.py]: yes, and yes

I guess if you want to be more direct you could use:

```python
with envcontext([('NO_COLOR', UNSET)]):
    ...
```

`envcotext` and `UNSET` come from `pre_commit.envcontext`

Comment (geieredgar) [reply tests/color_test.py]: Ok, I will change that because otherwise the coverage is reduced.

**Comment by asottile** [comment]
thanks again! and thanks @saper for the issue!

**Pull request merged**